### PR TITLE
Use Zen for AMD name, add Zen2 arch flags

### PR DESCRIPTION
**Pull request by jjwilke**
* Adds Zen as arch name, which is more exact architecture name than Epyc
* Adds Zen2 arch flags
* Removes Ryzen, which never added flags anywere

### CODE REVIEW

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -51,7 +51,8 @@ KOKKOS_ARCH_OPTION(PASCAL61        GPU  "NVIDIA Pascal generation CC 6.1")
 KOKKOS_ARCH_OPTION(VOLTA70         GPU  "NVIDIA Volta generation CC 7.0")
 KOKKOS_ARCH_OPTION(VOLTA72         GPU  "NVIDIA Volta generation CC 7.2")
 KOKKOS_ARCH_OPTION(TURING75        GPU  "NVIDIA Turing generation CC 7.5")
-KOKKOS_ARCH_OPTION(EPYC            HOST "AMD Epyc architecture")
```
Comment (dalg24) [inline cmake/kokkos_arch.cmake:54]: I feel like we need to ask ourselves: "Is it OK to just drop the option?"

Should we look for it and stop generation if it is provided?

Or is the warning issue by CMake good enough?
```
CMake Warning:
  Manually-specified variables were not used by the project:

    Kokkos_ARCH_EPYC
```

Comment (DavidPoliakoff) [reply cmake/kokkos_arch.cmake:54]: Perhaps something to the effect of a `KOKKOS_DEPRECATED_ARCHITECTURES` list, if a user selects something from that list we give a message? I could get behind that idea

**Comment by crtrott** [comment]
Lets add a discussion item for Wednesday.

File: cmake/kokkos_arch.cmake (hunk 1 above)
Comment (dalg24) [reply cmake/kokkos_arch.cmake:54]: We decided this would be a CMake error

Comment (jjwilke) [reply cmake/kokkos_arch.cmake:54]: Pushed new commit with proposed error checking for removed options.

**Comment by jjwilke** [comment]
To avoid mixing unrelated items on a PR, I will fix the Skylake,Skylake-X issue on a separate PR

File: cmake/kokkos_functions.cmake [hunk 2]
```diff
@@ -799,3 +799,36 @@ ENDFUNCTION(COMPILER_SPECIFIC_DEFS)
...
+      LIST(APPEND ${KEY_LIST_NAME} ${CURRENT_KEY})
+    ELSE()
+      SET(${VAR_PREFIX}${CURRENT_KEY} ${ENTRY})
+      SET(PARSE_KEY ON)
+    ENDIF()
+  ENDFOREACH()
+ENDMACRO()
+
+FUNCTION(kokkos_check_deprecated_options)
+  kokkos_key_value_map(DEPRECATED_MSG_ DEPRECATED_LIST ${ARGN})
```
Comment (dalg24) [inline cmake/kokkos_functions.cmake:826]: function name should be uppercase for consistency

Comment (dalg24) [reply cmake/kokkos_functions.cmake:826]: @jjwilke Do you want to fix that or merge as is?

Comment (jjwilke) [reply cmake/kokkos_functions.cmake:826]: I can fix it.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Retest this please.

**Pull request merged**